spec/section_spec.rb

```diff
@@ -53,7 +53,9 @@ class PageWithSection < SitePrism::Page
       it 'should raise an ArgumentError' do
         class Page < SitePrism::Page
         end
-        expect { Page.section :incorrect_section, '.section' }.to raise_error ArgumentError, 'You should provide section class either as a block, or as the second argument'
+        expect do
```

Comment:
Can we rewrite this spec as `{}` syntax but then do trailing `.` for the two args (Should really pass `with_message` as well as separate arg)

Reply:
I thought of several ways.
I selected a method using `{}` syntax.

[Method 1] Selected

```
context 'second argument is not a class and no block given' do
  subject(:section) { Page.section(:incorrect_section, '.section') }

  it 'should raise an ArgumentError' do
    class Page < SitePrism::Page
    end
    expect { section }.to raise_error ArgumentError, 'You should provide section class either as a block, or as the second argument'
  end
end
```

[Method 2]

```
context 'second argument is not a class and no block given' do
  subject { -> { Page.section(:incorrect_section, '.section') } }

  it 'should raise an ArgumentError' do
    class Page < SitePrism::Page
    end
    is_expected.to raise_error ArgumentError, 'You should provide section class either as a block, or as the second argument'
  end
end
```